File: practica2 de auxiliatura/composicion  y agregacion/ejercicio1.py
class Habitacion:
    def __init__(self, nombre, tamano):
        self.nombre = nombre
        self.tamano = tamano  # en metros cuadrados
    def get_nombre(self):
        return self.nombre
    def get_tamano(self):
        return self.tamano
    def set_nombre(self, nombre):
        self.nombre = nombre
    def set_tamano(self, tamano):
        self.tamano = tamano
    def mostrar_info(self):
        print(f"Habitación: {self.nombre}, Tamaño: {self.tamano} m²")
class Casa:
    def __init__(self, direccion):
        self.direccion = direccion
        self.habitaciones = []
    def get_direccion(self):
        return self.direccion
    def set_direccion(self, direccion):
        self.direccion = direccion
    def agregar_habitacion(self, habitacion):
        self.habitaciones.append(habitacion)
    def mostrar_casa(self):
        print(f"Dirección: {self.direccion}")
        print("Habitaciones:")
        for hab in self.habitaciones:
            hab.mostrar_info()
habitacion1 = Habitacion("Dormitorio", 12)
habitacion2 = Habitacion("Sala", 20)
habitacion3 = Habitacion("Cocina", 10)
casa = Casa("Calle Falsa 123")
casa.agregar_habitacion(habitacion1)
casa.agregar_habitacion(habitacion2)
casa.agregar_habitacion(habitacion3)
casa.mostrar_casa()

File: practica2deAuxiliaturaPGRA2/src/AgregacionyComposicion/ejercicio1.java
package AgregacionyComposicion;
import java.util.ArrayList;
class Habitacion {
    private String nombre;
    private double tamano;
    public Habitacion(String nombre, double tamano) {
        this.nombre = nombre;
        this.tamano = tamano;
    }
    public String getNombre() {
        return nombre;
    }
    public double getTamano() {
        return tamano;
    }
    public void setNombre(String nombre) {
        this.nombre = nombre;
    }
    public void setTamano(double tamano) {
        this.tamano = tamano;
    }
    public void mostrarInfo() {
        System.out.println("Habitación: " + nombre + ", Tamaño: " + tamano + " m²");
    }
}
class Casa {
    private String direccion;
    private ArrayList<Habitacion> habitaciones;

    public Casa(String direccion) {
        this.direccion = direccion;
        this.habitaciones = new ArrayList<>();
    }

    public String getDireccion() {
        return direccion;
    }

    public void setDireccion(String direccion) {
        this.direccion = direccion;
    }

    public void agregarHabitacion(Habitacion h) {
        habitaciones.add(h);
    }
    public void mostrarCasa() {
        System.out.println("Dirección: " + direccion);
        System.out.println("Habitaciones:");
        for (Habitacion h : habitaciones) {
            h.mostrarInfo();
        }
    }
}
public class ejercicio1 {
    public static void main(String[] args) {
        Habitacion h1 = new Habitacion("Dormitorio", 12);
        Habitacion h2 = new Habitacion("Sala", 20);
        Habitacion h3 = new Habitacion("Cocina", 10);

        Casa casa = new Casa("Calle Falsa 123");
        casa.agregarHabitacion(h1);
        casa.agregarHabitacion(h2);
        casa.agregarHabitacion(h3);
        casa.mostrarCasa();
    }
}
